### PR TITLE
Testing for Question & Answer service layer

### PR DESCRIPTION
**Pull request by mattfreshwaters**
Application runs fine, as well as all tests pass ofc.  Some method names are kind of funky (i.e. getByQuestionQuestionId) - I can go ahead and refactor this later, it just means refactoring the model (questionId --> id) and a bunch of methods.

### CODE REVIEW

File: src/main/java/com/talentpath/shamazin/showItemPage/services/QuestionService.java [hunk 1]
```diff
@@ -14,11 +17,26 @@
 
     public List<Question> getAllQuestions() {return questionDao.findAll();}
 
-    public Question getQuestionById(Integer qId){ return questionDao.findById(qId).get(); }
+    public List<Question> getByItemFamilyId(Integer itemFamilyId){return questionDao.findByItemFamilyId(itemFamilyId);}
```
Comment (timluu) [inline src/main/java/com/talentpath/shamazin/showItemPage/services/QuestionService.java:20]: see overall comment

File: src/main/java/com/talentpath/shamazin/showItemPage/services/AnswerService.java [hunk 2]
```diff
@@ -15,11 +18,31 @@
 
     public List<Answer> getAllAnswers(){return repo.findAll();}
 
-    public Answer getAnswerById(Integer aId){return repo.findById(aId).get();}
+    public List<Answer> getByQuestionQuestionId(Integer questionId) {
```
Comment (timluu) [inline src/main/java/com/talentpath/shamazin/showItemPage/services/AnswerService.java:21]: see overall comment

**Review by timluu — APPROVED** [review]
overall, looks good to me. you already mentioned the naming thing, but i think it's fine given the model we have. the only thing i would say is the naming consistency on some parts (on the two lines i commented on). instead of "getByQuestionQuestionId" it could be "getAnswersByQuestionQuestionId" so that it makes clear from the get-go what we're getting. also because i think that's how the naming convention is in the rest of the code (e.g. "getAnswerById", "getAllAnswers", etc).

overall though, those are just nitpicks, and you should be good to go tbh

**Review by jaitken — APPROVED** [review]
Everything looks good to me, I agree with tim about the naming

**Comment by mattfreshwaters** [comment]
> overall, looks good to me. you already mentioned the naming thing, but i think it's fine given the model we have. the only thing i would say is the naming consistency on some parts (on the two lines i commented on). instead of "getByQuestionQuestionId" it could be "getAnswersByQuestionQuestionId" so that it makes clear from the get-go what we're getting. also because i think that's how the naming convention is in the rest of the code (e.g. "getAnswerById", "getAllAnswers", etc).
> 
> overall though, those are just nitpicks, and you should be good to go tbh

Ok gotcha, I see what you're saying, I'll adjust the names accordingly push up and merge

**Pull request merged**